traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/router/TrafficRouter.java

```diff
@@ -559,9 +563,47 @@ public HTTPRouteResult multiRoute(final HTTPRequest request, final Track track)
 		return routeResult;
 	}
 
-	public String buildPatternBasedHashString(final DeliveryService deliveryService, final String requestPath) {
+	/**
+	 * Creates a string to be used in consistent hashing
+	 *
+	 * This uses simply the request path by default, but will consider any and all Query Parameters
+	 * that are in deliveryService's {@link DeliveryService.consistentHashQueryParams} set as well.
+	 * It will also fall back on the request path if the query parameters are not UTF-8-encoded.
+	 *
+	 * @param deliveryService The Delivery Service being requested
+	 * @param request An {@link HTTPRequest} representing the client's request.
+	 * @return A string appropriate to use for consistent hashing to service the request
+	*/
+	@SuppressWarnings({"PMD.CyclomaticComplexity"})
+	public String buildPatternBasedHashString(final DeliveryService deliveryService, final HTTPRequest request) {
+		final String requestPath = request.getPath();
 		if (deliveryService.getConsistentHashRegex() != null && !deliveryService.getConsistentHashRegex().isEmpty() && !requestPath.isEmpty()) {
-			return buildPatternBasedHashString(deliveryService.getConsistentHashRegex(), requestPath);
+			final StringBuilder hashString = new StringBuilder(request.getPath());
+			final SortedSet<String> qparams = new TreeSet<String>();
+			try {
+				for (final String qparam : request.getQueryString().split("&")) {
+					if (qparam.isEmpty()) {
+						continue;
+					}
+
+					String[] parts = qparam.split("=");
+					for (short i = 0; i<parts.length; ++i) {
+						parts[i] = URLDecoder.decode(parts[i]);
```

Comment:
Apparently this method is deprecated because the method that specified the encoding should be used instead ("UTF-8" I believe?): https://docs.oracle.com/javase/8/docs/api/java/net/URLDecoder.html#decode-java.lang.String-java.lang.String-

Reply:
Oh yeah, I remember reading that. I don't think you can just give it literally `"UTF-8"` though, because it failed to compile I think... You're supposed to give it some kind of `Charset` object I think. I'll try to figure that out.

Reply:
No, I'm mistaken. Must've misspelled it or something else was wrong.